src/bot.ts

```diff
@@ -22,6 +22,7 @@ const MIN_PRESENCE_UPDATE_DELAY = 250;
 // TODO: This is bad. We should be serving the icon from the own homeserver.
 const MATRIX_ICON_URL = "https://matrix.org/_matrix/media/r0/download/matrix.org/mlxoESwIsTbJrfXyAAogrNxA";
 class ChannelLookupResult {
+  public guild: Discord.Guild;
```

Comment:
Channels *have* guilds so I don't really like including it in this. You can get the guild of a channel with ``channel.guild``

Reply:
You ***WISH*** you could, but nah https://discord.js.org/#/docs/main/stable/class/Channel

Unless the docs are wrong/outdated.

Reply:
Ah...see this is where I cheated and casted everything to a TextChannel. Given we only ever operate on TextChannels, it's a safe bet for now.